### PR TITLE
feat: Trial and Task log filters [WEB-239] [WEB-240]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Trial and task logs can have an agent, container, rank, etc. which we can use to filter these logs. When a log entry doesn't have a rank, the **current** version sees a NULL and filters it out on the server.  The returned agent/container/rank values are used to populate log filters.

In this update:
- the null value is included in API responses as an empty string.
- rankIds accepts and array of strings (to include empty string) instead of ints
- "No Container" works by sending an empty string which is the default for these logs. rankId has a number or null value so the empty string cannot be used. Query with the new `FilterOperationInOrNull`

## Test Plan

- Launch a Jupyter notebook locally and copy its task ID (such as 926e5c59-afd8-415d-b3a4-89846d194dee).
- Visit `http://localhost:8080/api/v1/tasks/:task_id/logs/fields` - confirm each array has at least one value, and it is an empty string, no NULLs, no empty values, no ints for rank.
- Run an experiment and copy its trial ID (an int, **not** the experiment ID)
- Visit `http://localhost:8080/api/v1/trials/:trial_id/logs/fields` - confirm each array has at least one value, and it is an empty string, no NULLs, no empty values, no ints for rank.
- For dropdowns: on a trial logs page, you need more than one option for a dropdown to appear. You probably will see an Agents dropdown with No Agent.
- For trial logs, you can check rankId is working by going to a trial log endpoint with rankIds set to a blank value, such as `http://localhost:8080/api/v1/trials/:trial_id/logs?rankIds=`

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by gt2345 — APPROVED** [review]
Works great~

File: harness/determined/common/api/bindings.py [hunk 1]
```diff
@@ -9593,7 +9593,7 @@ class v1TaskLogsFieldsResponse:
     agentIds: "typing.Optional[typing.Sequence[str]]" = None
     allocationIds: "typing.Optional[typing.Sequence[str]]" = None
     containerIds: "typing.Optional[typing.Sequence[str]]" = None
-    rankIds: "typing.Optional[typing.Sequence[int]]" = None
```
Comment (stoksc) [inline harness/determined/common/api/bindings.py:9596]: rank ID _is_ an int. changing the API to a string to accomplish this seems like a strange choice

File: master/internal/api_tasks.go [hunk 2]
```diff
@@ -429,10 +430,33 @@ func constructTaskLogsFilters(req *apiv1.TaskLogsRequest) ([]api.Filter, error)
 		}
 	}
 
-	addInFilter("allocation_id", req.AllocationIds, len(req.AllocationIds))
-	addInFilter("agent_id", req.AgentIds, len(req.AgentIds))
+	addNumFilter := func(field string, values []string) {
+		if values != nil {
+			if slices.Contains(values, "") {
```
Comment (stoksc) [inline master/internal/api_tasks.go:435]: why are we assuming `values` would only contain one `""` entry? i dont think the api enforces that?

File: master/internal/api_tasks.go [hunk 3]
```diff
@@ -429,10 +430,33 @@ func constructTaskLogsFilters(req *apiv1.TaskLogsRequest) ([]api.Filter, error)
 		}
 	}
 
-	addInFilter("allocation_id", req.AllocationIds, len(req.AllocationIds))
-	addInFilter("agent_id", req.AgentIds, len(req.AgentIds))
+	addNumFilter := func(field string, values []string) {
```
Comment (stoksc) [inline master/internal/api_tasks.go:433]: this seems like "add num filter" and more like "add cleaned in filter"

File: master/internal/elastic/elastic_task_logs.go [hunk 4]
```diff
@@ -488,10 +488,10 @@ type intAggResult struct {
 	} `json:"buckets"`
 }
 
-func (r intAggResult) toKeysInt32() []int32 {
-	var keys []int32
+func (r intAggResult) toKeysInt32() []string {
```
Comment (stoksc) [inline master/internal/elastic/elastic_task_logs.go:491]: `toKeysString`?

File: master/static/srv/get_task_logs_fields.sql [hunk 5]
```diff
@@ -1,6 +1,6 @@
 SELECT
     array_to_json(
-        array_remove(
+        array_replace(
```
Comment (stoksc) [inline master/static/srv/get_task_logs_fields.sql:3]: i think the whole approach here is a little off - we converted everything to strings and used `""` to indicate null.. seems to make a lot more sense _not_ to convert any types and just let null but null, but not filter it?

**Review by stoksc — COMMENTED** [review]
posthumous review, hope that is ok. i didn't get a chance to see/know about this one earlier.

File: master/internal/api_tasks.go (hunk 2 above)
Comment (mapmeld) [reply master/internal/api_tasks.go:435]: I don't think it's necessary to remove the empty string from `FilterOperationInOrNull` so I will remove this concern

File: master/internal/api_tasks.go (hunk 3 above)
Comment (mapmeld) [reply master/internal/api_tasks.go:433]: going with `addNullInclusiveFilter`

**Pull request merged**